Loop/Managers/LoopDataManager.swift

```diff
@@ -1896,23 +1842,32 @@ extension LoopDataManager {
         guard FeatureFlags.simulatedCoreDataEnabled else {
             fatalError("\(#function) should be invoked only when simulated core data is enabled")
         }
+        
+        guard let doseStore = doseStore as? DoseStore else {
```

Comment:
[nit] I would copy the pattern you use above in `generateSimulatedHistoricalCoreData` to check all of the stores at the top of the function and call `fatalError` once if any fail.